### PR TITLE
fix(ACLStorageWrapper): Fix return types

### PR DESCRIPTION
**Pull request by provokateurin**
Updates of https://github.com/nextcloud/server/pull/48219 and subsequent fixes.

### CODE REVIEW

**Comment by provokateurin** [comment]
(Merging to fix CI on master)

File: lib/ACL/ACLStorageWrapper.php [hunk 1]
```diff
@@ -291,16 +291,20 @@ public function getETag($path): string|false {
...
 	}
 
-	public function getDirectoryContent($directory): \Traversable {
-		foreach ($this->getWrapperStorage()->getDirectoryContent($directory) as $data) {
+	public function getDirectoryContent($directory): \Traversable|false {
+		$content = $this->getWrapperStorage()->getDirectoryContent($directory);
+		if ($content === false) {
+			return false;
+		}
+		foreach ($content as $data) {
```
Comment (come-nc) [inline lib/ACL/ACLStorageWrapper.php:307]: Still puzzled by this return false thing :thinking:

**Comment by come-nc** [comment]
(Approved, but github will not let me approve after merge)

File: lib/ACL/ACLStorageWrapper.php (hunk 1 above)
Comment (provokateurin) [reply lib/ACL/ACLStorageWrapper.php:307]: I'll respond in the server PR

**Pull request merged**